react-client-app/src/components/UploadScanReport.jsx

```diff
@@ -0,0 +1,398 @@
+import React, { useState, useEffect, useRef } from 'react'
+import PageHeading from './PageHeading'
+import {
+    Select, Box, Text, Button, Flex, Spinner, Container, Input, Tooltip, CloseButton, ScaleFade, useDisclosure, Switch,
+    FormControl, FormLabel, FormErrorMessage
+} from "@chakra-ui/react"
+import { useGet, usePost } from '../api/values'
+import ToastAlert from './ToastAlert'
+import ConceptTag from './ConceptTag'
+
+const UploadScanReport = ({ setTitle }) => {
+
+    const { isOpen, onOpen, onClose } = useDisclosure()
+    const [alert, setAlert] = useState({ hidden: true, title: '', description: '', status: 'error' });
+    const [dataPartners, setDataPartners] = useState([{ name: "------" }]);
+    const [selectedDataPartner, setselectedDataPartner] = useState({ name: "------" });
+
+    const [datasets, setDatasets] = useState([{ name: "------" }]);
+    const [loadingDataset, setLoadingDataset] = useState(false);
+    const [addingDataset, setAddingDataset] = useState(false);
+    const [projectVisibleToPublic, setProjectVisibleToPublic] = useState(true);
+    const [projects, setProjects] = useState([]);
+    const [projectList, setProjectList] = useState(undefined);
+
+    const [users, setUsers] = useState([]);
+    const [usersList, setUsersList] = useState(undefined);
+
+    const [addDatasetMessage, setAddDatasetMessage] = useState(null);
+    const [formErrors, setFormErrors] = useState({});
+
+    const [selectedDataset, setselectedDataset] = useState({ name: "------" });
+
+    const scanReportName = useRef();
+    const createDatasetRef = useRef();
+    const [whiteRabbitScanReport, setWhiteRabbitScanReport] = useState(null);
+    const [dataDictionary, setDataDictionary] = useState(null);
+    const [loadingMessage, setLoadingMessage] = useState("Loading page")
+    const [uploadLoading, setUploadLoading] = useState(false)
+    useEffect(async () => {
+        setTitle(null)
+        const dataPartnerQuery = await useGet("/datapartners/")
+        setDataPartners([{ name: "------" }, ...dataPartnerQuery])
+        setLoadingMessage(null)
+        const projectsQuery = await useGet("/projects/")
+        setProjectList(projectsQuery)
+        const usersQuery = await useGet("/users/")
+        setUsersList(usersQuery)
+    }, []);
+
+    useEffect(async () => {
+        setFormErrors({ ...formErrors, datapartner: undefined })
+        // change the dataset list
+        setLoadingDataset(true)
+        const dataPartnerId = selectedDataPartner.id
+        if (dataPartnerId != undefined) {
+            const datasets_query = await useGet(`/datasetsfilter/?data_partner=${dataPartnerId}`)
```

Comment:
after merging #392, you can change this to:
```suggestion
            const datasets_query = await useGet(`/datasets/?data_partner=${dataPartnerId}`)
```